### PR TITLE
examples : Enhanced compatibility with older Android versions using Java

### PR DESCRIPTION
**Pull request by litongjava**
~~Added support for older versions of Android of Java~~
Enhanced compatibility with older Android versions using Java.

### CODE REVIEW

**Comment by bobqianic** [comment]
Great job! Could you add a simple README so everyone will know how to use this example?

**Comment by litongjava** [comment]
Please Check

**Comment by litongjava** [comment]
I'm done. Please review the code and merge it.

File: bindings/java/build.gradle [hunk 1]
```diff
@@ -7,7 +7,7 @@ plugins {
 
 archivesBaseName = 'whispercpp'
 group = 'io.github.ggerganov'
-version = '1.4.0'
+version = '1.4.1'
```
Comment (bobqianic) [inline bindings/java/build.gradle:10]: Why do we need to update the binding version number?

File: examples/whisper.android.java/app/src/main/assets/logback.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
+      <maxFileSize>10MB</maxFileSize>
+    </triggeringPolicy>
+  </appender>
+  <!-- 日志输出级别 和输出源 -->
+  <root level="info">
+    <appender-ref ref="STDOUT" />
+    <appender-ref ref="FILE" />
+  </root>
+</configuration>
```
Comment (bobqianic) [inline examples/whisper.android.java/app/src/main/assets/logback.xml:40]: What's the purpose of this?

File: examples/whisper.android.java/app/src/main/java/com/litongjava/whisper/android/java/services/WhisperService.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  @RequiresApi(api = Build.VERSION_CODES.O)
+  public void transcribeSample(TextView tv, File sampleFile) {
+    String msg = "";
+    msg = "transcribe file from :" + sampleFile.getAbsolutePath();
+    outputMsg(tv, msg);
+
+    Long start = System.currentTimeMillis();
+    float[] audioData = new float[0];  // 读取音频样本
```
Comment (bobqianic) [inline examples/whisper.android.java/app/src/main/java/com/litongjava/whisper/android/java/services/WhisperService.java:54]:
```suggestion
    float[] audioData = new float[0];  // Read audio samples
```

File: bindings/java/build.gradle (hunk 1 above)
Comment (litongjava) [reply bindings/java/build.gradle:10]: I chaged that project io.github.ggerganov whispercpp

File: examples/whisper.android.java/app/src/main/assets/logback.xml (hunk 2 above)
Comment (litongjava) [reply examples/whisper.android.java/app/src/main/assets/logback.xml:40]: this is logback config file,In my project I use logback to recored log.

**Comment by litongjava** [comment]
I've update the logback file,change Chinese to English.

**Comment by litongjava** [comment]
Thanks for you feedback if you have other question please let me know.

File: bindings/java/build.gradle (hunk 1 above)
Comment (bobqianic) [inline bindings/java/build.gradle:10]:
```suggestion
version = '1.4.0'

```

File: .gitignore [hunk 4]
```diff
@@ -48,3 +48,8 @@ bindings/java/.idea/
 .idea/
 
 benchmark_results.csv
+cmake-build-debug/
+*.bin
```
Comment (bobqianic) [inline .gitignore:52]:
```suggestion
```

**Review by bobqianic — CHANGES_REQUESTED** [review]

**Review by bobqianic — APPROVED** [review]

**Comment by litongjava** [comment]
Thank you

**Comment by ggerganov** [comment]
We would need to add a build + test if possible to the CI, otherwise the example will get outdated very quickly

**Comment by litongjava** [comment]
When do you plan to merge this PR?

**Comment by litongjava** [comment]
I have added the github CI, please review the code and merge this PR

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -396,6 +396,28 @@ jobs:
...
+    steps:
+      - name: Clone
+        uses: actions/checkout@v3
+
+      - name: Install Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: zulu
+          java-version: '8'
+          cache: gradle'
```
Comment (ggerganov) [inline .github/workflows/build.yml:411]: Missing or extra quote on this line

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -403,11 +403,12 @@ jobs:
...
         uses: actions/checkout@v3
 
-      - name: Install Java
+      - name: set up JDK 8
         uses: actions/setup-java@v3
         with:
-          distribution: zulu
           java-version: '8'
+          distribution: 'temurin'
+          cache: gradle
```
Comment (bobqianic) [inline .github/workflows/build.yml:411]: Why do we have to stick with Java 8? I just checked the error message, and it looks like the minimum Java version required is Java 11 (55.0).
 https://github.com/ggerganov/whisper.cpp/actions/runs/6789530038/job/18456933658?pr=1382#step:4:9

File: examples/whisper.android.java/app/src/main/jni/whisper/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+set(CMAKE_CXX_STANDARD 11)
+set(WHISPER_LIB_DIR ${CMAKE_SOURCE_DIR}/../../../../../../../)
+
+set(
+        SOURCE_FILES
+        ${WHISPER_LIB_DIR}/ggml.c
+        ${WHISPER_LIB_DIR}/ggml-alloc.c
+        ${WHISPER_LIB_DIR}/whisper.cpp
+        ${CMAKE_SOURCE_DIR}/jni.c
+)
```
Comment (bobqianic) [inline examples/whisper.android.java/app/src/main/jni/whisper/CMakeLists.txt:14]:
```suggestion
set(
        SOURCE_FILES
        ${WHISPER_LIB_DIR}/ggml.c
        ${WHISPER_LIB_DIR}/ggml-alloc.c
        ${WHISPER_LIB_DIR}/ggml-backend.c
        ${WHISPER_LIB_DIR}/ggml-quants.c
        ${WHISPER_LIB_DIR}/whisper.cpp
        ${CMAKE_SOURCE_DIR}/jni.c
)
```

**Review by bobqianic — CHANGES_REQUESTED** [review]

**Comment by litongjava** [comment]
android_java of CI runs successfully in github action, Please review code

**Comment by litongjava** [comment]
@nalbion Please review code

File: examples/whisper.android.java/local.properties
Comment (ggerganov) [inline examples/whisper.android.java/local.properties:1]: This file should not be source controlled

**Comment by litongjava** [comment]
Thanks for you reivew and feedback.
1.local.properties has removed from git 2.assert test code is enabled 3.Please review code

**Pull request merged**